pkg/proxy/winkernel/proxier.go

```diff
@@ -479,19 +505,73 @@ func NewProxier(
 	}
 
 	healthChecker := healthcheck.NewServer(hostname, recorder, nil, nil) // use default implementations of deps
+	var hns HostNetworkService
+	hns = hnsV1{}
 
 	// TODO : Make this a param
 	hnsNetworkName := os.Getenv("KUBE_NETWORK")
 	if len(hnsNetworkName) == 0 {
-		return nil, fmt.Errorf("Environment variable KUBE_NETWORK not initialized")
+		klog.V(3).Infof("Environment variable KUBE_NETWORK not initialized")
+		hnsNetworkName = config.NetworkName
+		if len(hnsNetworkName) == 0 {
+			return nil, fmt.Errorf("network-name flag not set")
+		}
 	}
-	hnsNetwork, err := getHnsNetworkInfo(hnsNetworkName)
+
+	hnsNetwork, err := hcsshim.GetHNSNetworkByName(hnsNetworkName)
 	if err != nil {
-		klog.Fatalf("Unable to find Hns Network specified by %s. Please check environment variable KUBE_NETWORK", hnsNetworkName)
+		klog.Fatalf("Unable to find Hns Network specified by %s. Please check environment variable KUBE_NETWORK or network-name flag", hnsNetworkName)
 		return nil, err
 	}
+	hnsNetworkInfo, err := getHnsNetworkInfo(hnsNetworkName)
 
 	klog.V(1).Infof("Hns Network loaded with info = %v", hnsNetwork)
+	var isDSR bool
+	isDSR = config.EnableDSR
+
+	var sourceVip string
+	var hostMac string
+	if hnsNetwork.Type == "Overlay" {
+		supportedFeatures := hcn.GetSupportedFeatures() //TODO Create RemoteSubnet Feature
+		if supportedFeatures.Api.V2 {
+			hns = hnsV2{}
```

Comment:
HNS v2 is required for the DSR feature, which is required (by Calico) with the l2bridge network type.  Can this check be moved out of the Type == Overlay conditional?

Reply:
Looks like the blocking PR has now been merged? https://github.com/Microsoft/hcsshim/pull/436